util/ranger/detacher.go

```diff
@@ -296,6 +296,49 @@ func (d *rangeDetacher) detachCNFCondAndBuildRangeForIndex(conditions []expressi
 		if eqOrInCount > 0 {
 			newCols := d.cols[eqOrInCount:]
 			newLengths := d.lengths[eqOrInCount:]
+			saveConditions := newConditions
```

Comment:
We want to distinguish the saveConditions and newConditions?(If we change one, the other one is unchanged?)
If so, try
```go
saveConditions := make([]Expression, len(newConditions))
copy(saveConditions, newConditions)
```

Reply:
fixed